File: content/problems/0120-triangle/solutions/bottom-up-dp/Solution.java
class Solution {
    public int minimumTotal(List<List<Integer>> triangle) { int[] dp = new int[triangle.size() + 1]; for (int row = triangle.size() - 1; row >= 0; row--) for (int column = 0; column <= row; column++) dp[column] = triangle.get(row).get(column) + Math.min(dp[column], dp[column + 1]); return dp[0]; }
}

File: content/problems/0120-triangle/solutions/bottom-up-dp/Solution.kt
class Solution {
    fun minimumTotal(triangle: List<List<Int>>): Int { val dp = IntArray(triangle.size + 1); for (row in triangle.lastIndex downTo 0) for (column in 0..row) dp[column] = triangle[row][column] + minOf(dp[column], dp[column + 1]); return dp[0] }
}
